### DISCUSSION TITLE
Reading a date

### DISCUSSION THREAD

**Discussion by TonioIncredible** [PHPOffice/PhpSpreadsheet#3115] [Q&A] [answered]
Good morning,
I am reading an Excel sheet which contains dates. When I open the file with Excel, I see dates, but when I import the file with PHPSpreadSheet, sometimes I get a date (like "23/03/1970") or sometimes I get a number representing a date (like "25940" instead of "07/01/1971").

How can I systematically get a date?

Thanks for your answer
Antoine

**Comment by MarkBaker** [comment, marked as answer]
I can see that I'm going to have to rewrite the [documentation](https://phpspreadsheet.readthedocs.io/en/latest/topics/recipes/#write-a-date-or-time-into-a-cell) to be very explicit about how dates work in MS Excel; and explain how to read dates as well as write them. [This article](https://www.excelcampus.com/functions/how-dates-work-in-excel/) explains in more detail about dates in MS Excel.

TLDR: MS Excel maintains dates as a serialized timestamp, a numeric value that represents the number of days since a base start date of 1/1/1900 (or 1/1/1904 if the file was created using the Apple Mac version of Microsoft Office). This is similar to a unix timestamp, a numeric value representing a date.

If you get a value of "23/03/1970", then you're getting a string; just because it looks like a date to a human, it's still a string. if you're getting a value of 25940, that could be a date, or simply a number. The only difference between a date and a number is the number format mask.

If you know that a cell should contain a date, then the Cell's `getFormattedValue()` method should return that value as a human-readable formatted string, similar to "07/01/1971" (although the exact formatting will depend on the formatting mask).

The `PhpOffice\PhpSpreadsheet\Shared\Date` class provides a series of methods to convert between Excel timestamps and unix timestamp, PHP DateTime object, and formatted strings (and vice versa).

PhpSpreadsheet also provides a method to determine if a cell value is a number or a date
```php
PhpOffice\PhpSpreadsheet\Shared\Date::isDateTime($worksheet->getCell('A1'));
```

**Comment by TonioIncredible** [comment]
Thanks, 
I will try that.
